contracts/0.8.9/LidoOracleNew.sol

```diff
@@ -0,0 +1,571 @@
+    /// - N after calling initialize() during deployment from scratch, where N is the current contract version
+    /// - N after upgrading contract from the previous version (after calling finalize_vN())
+    bytes32 internal constant CONTRACT_VERSION_POSITION = keccak256("lido.LidoOracle.contractVersion");
+
+    /// Receiver address to be called when the report is pushed to Lido
+    bytes32 internal constant BEACON_REPORT_RECEIVER_POSITION = keccak256("lido.LidoOracle.beaconReportReceiver");
+
+    /// Upper bound of the reported balance possible increase in APR, controlled by the governance
+    bytes32 internal constant ALLOWED_BEACON_BALANCE_ANNUAL_RELATIVE_INCREASE_POSITION =
+        keccak256("lido.LidoOracle.allowedBeaconBalanceAnnualRelativeIncrease");
+
+    /// Lower bound of the reported balance possible decrease, controlled by the governance
+    ///
+    /// @notice When slashing happens, the balance may decrease at a much faster pace. Slashing are
+    /// one-time events that decrease the balance a fair amount - a few percent at a time in a
+    /// realistic scenario. Thus, instead of sanity check for an APR, we check if the plain relative
+    /// decrease is within bounds.  Note that it's not annual value, its just one-jump value.
+    bytes32 internal constant ALLOWED_BEACON_BALANCE_RELATIVE_DECREASE_POSITION =
+        keccak256("lido.LidoOracle.allowedBeaconBalanceDecrease");
+
+
+    ///! STRUCTURED STORAGE OF THE CONTRACT
+    ///! Inherited from CommitteeQuorum:
+    ///! SLOT 0: address[] members
+    ///! SLOT 1: bytes[] distinctReports
+    ///! SLOT 2: bytes[] distinctReportHashes
+    ///! SLOT 3: bytes32[] distinctReportCounters
+    ///! Inherited from AccessControlEnumerable:
+    ///! SLOT 4: mapping(bytes32 => RoleData) _roles
+    ///! SLOT 5: mapping(bytes32 => EnumerableSet.AddressSet) _roleMembers
+
+
+    /**
+     * @notice Initialize the contract (version 3 for now) from scratch
+     * @dev For details see https://github.com/lidofinance/lido-improvement-proposals/blob/develop/LIPS/lip-10.md
+     * @param _admin Admin which can modify OpenZeppelin role holders
+     * @param _lido Address of Lido contract
+     * @param _epochsPerFrame Number of epochs per frame
+     * @param _slotsPerEpoch Number of slots per epoch
+     * @param _secondsPerSlot Number of seconds per slot
+     * @param _genesisTime Genesis time
+     * @param _allowedBeaconBalanceAnnualRelativeIncrease Allowed beacon balance annual relative increase (e.g. 1000 means 10% increase)
+     * @param _allowedBeaconBalanceRelativeDecrease Allowed beacon balance instantaneous decrease (e.g. 500 means 5% decrease)
+     */
+    function initialize(
+        address _admin,
+        address _lido,
+        uint64 _epochsPerFrame,
+        uint64 _slotsPerEpoch,
+        uint64 _secondsPerSlot,
+        uint64 _genesisTime,
+        uint256 _allowedBeaconBalanceAnnualRelativeIncrease,
```

Comment:
Sanity checks should be re-worked in accordance to https://hackmd.io/R6SKBwd7Tke0d7yg8R_mwg#Safety-nets